src/inc/utilcode.h

```diff
@@ -134,6 +134,17 @@ inline TADDR PCODEToPINSTR(PCODE pc)
 #endif
 }
 
+// Convert from a PINSTR to the corresponding PCODE.  On many architectures this will be the identity function;
+// on ARM, this will raise the THUMB bit.
+inline PCODE PINSTRToPCODE(TADDR addr)
+{
+#ifdef _TARGET_ARM_
+    return DataPointerToThumbCode<TADDR,PCODE>(addr);
```

Comment:
You have the template argument types in a wrong order here. The first argument is the return type, the second is the source type.